src/transforms/coercer.rs

```diff
@@ -56,11 +56,12 @@ impl Transform for Coercer {
             // below, as it will be fewer steps to fully recreate the
             // event than to scan the event for extraneous fields after
             // conversion.
-            let drain = log.drain().collect::<Vec<_>>();
-            for (field, value) in drain {
+            let mut new_event = Event::new_empty_log();
```

Comment:
this seems quite unfortunate, do you think there may be otherways around this? It looks like the previous version was draining and moving while this one copies.

Reply:
Although `drain` is not implemented for `BTreeMap`, it turned out to be possible to avoid cloning using `into_iter` (7db3a12de1f89b4fa9b8423f076785e6ef011492).